lib/rucio/api/rule.py

```diff
@@ -228,7 +230,7 @@ def delete_replication_rule(rule_id, purge_replicas, issuer, vo='def', session=N
 
 
 @transactional_session
-def update_replication_rule(rule_id, options, issuer, vo='def', session=None):
+def update_replication_rule(rule_id: str, options: Dict[str, Any], issuer: str, vo='def', session=None):
```

Comment:
Why avoid type hints for `vo` and `session`?

Reply:
in the case of vo, mypy can infer the type from the default value given. i consider it to be redundant information to add a `str` tye-hint.
in the case of `session`, Radu is working on this already https://github.com/rcarpa/rucio/commit/94bcc0940451545628e0e53f5feb324d7911c665

Reply:
For type hints in parameters, I find that ‘explicit is better than implicit’, as instructed by _The Zen of Python_. But I won’t push you.

Reply:
you are right. consider it done